lib/plugins/aws/invoke-local/index.js

```diff
@@ -543,6 +543,7 @@ class AwsInvokeLocal {
       configuredEnvVars,
       envVarsFromOptions
     );
+    // #7747, envVars is not Array, and so we cannot use Array.prototype.flatMap() here
     const envVarsDockerArgs = _.flatMap(envVars, (value, key) => ['--env', `${key}=${value}`]);
```

Comment:
Maybe we could still replace it with different calls, e.g. `Object.entries` on `envVars` to change it into an array?

Reply:
That worked, though note this small change:

```
const envVarsDockerArgs = _.flatMap(envVars, (value, key) => ['--env', `${key}=${value}`]);
```

to

```
const envVarsDockerArgs = Object.entries(envVars).flatMap( (currentValue) => ['--env', `${currentValue[0]}=${currentValue[1]}`])
```

`array.flatMap()` passes in the values from `entries()` as a single array (`currentValue`). So, `key` and `value` are then indices in the `currentValue` array.

see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap#syntax

Reply:
Thanks for clarification, I think that works well and allows to replace `_.flatMap` 👍